include/exadg/solvers_and_preconditioners/preconditioners/preconditioner_amg.h

```diff
@@ -110,6 +112,148 @@ class PreconditionerAMG : public PreconditionerBase<TrilinosNumber>
 
   AMGData amg_data;
 };
+
+
+/*
+ * Wrapper class for BoomerAMG from Hypre
+ */
+template<typename Operator, typename PreconditionerNumber>
+class PreconditionerBoomerAMG : public PreconditionerBase<PreconditionerNumber>
+{
+private:
+  typedef LinearAlgebra::distributed::Vector<PreconditionerNumber> VectorType;
+
+#ifdef DEAL_II_WITH_PETSC
+public:
+  // distributed sparse system matrix
+  PETScWrappers::MPI::SparseMatrix system_matrix;
+
+private:
+  PETScWrappers::PreconditionBoomerAMG amg;
+#endif
+
+public:
+  PreconditionerBoomerAMG(Operator const & op, AMGData data = AMGData())
+    : pde_operator(op), amg_data(data)
+  {
+#ifdef DEAL_II_WITH_PETSC
+    // initialize system matrix
+    pde_operator.init_system_matrix(system_matrix);
+
+    // calculate_matrix
+    pde_operator.calculate_system_matrix(system_matrix);
+
+    // initialize the Boomer AMG data structures, translate from Trilinos
+    // settings; right now we must skip most parameters because there does not
+    // appear to be a setting available in deal.II
+    PETScWrappers::PreconditionBoomerAMG::AdditionalData boomer_data;
+    boomer_data.symmetric_operator               = amg_data.data.elliptic;
+    boomer_data.strong_threshold                 = 0.25;
+    boomer_data.aggressive_coarsening_num_levels = 2;
+    amg.initialize(system_matrix, boomer_data);
+#else
+    AssertThrow(false, ExcMessage("deal.II is not compiled with PETSc!"));
+#endif
```

Comment:
Could you shift this part into a private member function `calculate_system_matrix()` to avoid code duplication?